Src/ILGPU/Runtime/ExchangeBuffer.cs

```diff
@@ -163,37 +243,23 @@ protected override void Dispose(bool disposing)
         /// The internally allocated CPU memory.
         /// </summary>
         [DebuggerBrowsable(DebuggerBrowsableState.Never)]
-        private readonly ViewPointerWrapper cpuMemory;
+        protected readonly ViewPointerWrapper cpuMemory;
 
         /// <summary>
         /// A cached version of the CPU memory pointer.
         /// </summary>
         [DebuggerBrowsable(DebuggerBrowsableState.Never)]
-        private readonly void* cpuMemoryPointer;
+        protected readonly void* cpuMemoryPointer;
```

Comment:
You can create a `protected` property to access the `cpuMemoryPointer`. Alternatively, you can convert the pointer into an auto-implemented `readonly` property.

Reply:
Having them anything more visible than internal is what yields the "Do not declare visible instance fields" problem. [Microsoft](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1051?view=vs-2019) suggests making them private and using a property, so I guess that's what I'll do. Is that what you were saying? Sorry, I didn't quite understand.

Reply:
Anyways, I just did it with a protected property.